src/test/java/teammates/test/cases/logic/StudentsLogicTest.java

```diff
@@ -547,53 +567,55 @@ public void testEnrollLinesChecking() throws Exception {
 
         enrollLines = headerLine + Const.EOL + lineWithCorrectInput + Const.EOL + lineWithCorrectInput;
 
-        invalidInfo = getInvalidityInfoInEnrollLines(enrollLines, courseId);
+        try {
+            studentsLogic.createAndValidateStudents(enrollLines, courseId);
+            signalFailureToDetectException();
+        } catch (EnrollException e) {
+            invalidInfoString = e.getMessage();
+        }
 
-        assertEquals(1, invalidInfo.size());
+        expectedInvalidInfoString = "Same email address as the student in line \"" + lineWithCorrectInput + "\"";
+        AssertHelper.assertContains(expectedInvalidInfoString, invalidInfoString);
 
         ______TS("enrollLines with a mix of all above cases");
         enrollLines = headerLine + Const.EOL + lineWithInvalidTeamName + Const.EOL
                 + lineWithInvalidTeamNameAndStudentNameAndEmail
                 + Const.EOL + lineWithExtraParameters + Const.EOL
                 + lineWithTeamNameEmpty + Const.EOL + lineWithCorrectInput + Const.EOL + "\t";
 
-        invalidInfo = getInvalidityInfoInEnrollLines(enrollLines, courseId);
+        try {
+            studentsLogic.createAndValidateStudents(enrollLines, courseId);
+            signalFailureToDetectException();
+        } catch (EnrollException e) {
+            invalidInfoString = e.getMessage();
+        }
```

Comment:
This block of code is repeated many times. Can extract as a utility method? e.g. `getErrorMessageFromEnrollAttempt`